File: Components/RewriteMe.Domain/Transcription/FileItem.cs
using System;
using System.Collections.Generic;
using RewriteMe.Domain.Enums;

namespace RewriteMe.Domain.Transcription
{
    public class FileItem
    {
        public FileItem()
        {
            RecognitionState = RecognitionState.None;
        }

        public Guid Id { get; set; }

        public Guid UserId { get; set; }

        public Guid ApplicationId { get; set; }

        public string Name { get; set; }

        public string FileName { get; set; }

        public string Language { get; set; }

        public RecognitionState RecognitionState { get; set; }

        public string OriginalSourceFileName { get; set; }

        public string SourceFileName { get; set; }

        public StorageSetting Storage { get; set; }

        public UploadStatus UploadStatus { get; set; }

        public TimeSpan TotalTime { get; set; }

        public TimeSpan TranscribedTime { get; set; }

        public DateTime DateCreated { get; set; }

        public DateTime? DateProcessedUtc { get; set; }

        public DateTime DateUpdatedUtc { get; set; }

        public bool IsDeleted { get; set; }

        public bool WasCleaned { get; set; }

        public bool CanMigrate => RecognitionState != RecognitionState.Converting ||
                                  RecognitionState != RecognitionState.InProgress;

        public IEnumerable<TranscribeItem> TranscribeItems { get; set; }
    }
}

File: WebControlPanelClient/src/app/_models/file-item.ts
import { RecognitionState } from '../_enums/recognition-state';
import { StorageSetting } from '../_enums/storage-setting';
import { UploadStatus } from '../_enums/upload-status';

export class FileItem {
    public id: string;
    public userId: string;
    public applicationId: string;
    public name: string;
    public fileName: string;
    public language: string;
    public recognitionState: RecognitionState;
    public originalSourceFileName: string;
    public originalSourcePath: string;
    public sourceFileName: string;
    public sourcePath: string;
    public originalContentType: string;
    public storage: StorageSetting;
    public uploadStatus: UploadStatus;
    public totalTime: any;
    public transcribedTime: string;
    public dateCreated: Date;
    public dateProcessed: Date;
    public dateUpdated: Date;
    public isDeleted: boolean;
    public isPermanentlyDeleted: boolean;
    public wasCleaned: boolean;
}
